### PR TITLE
LoadFragment2

### PR DESCRIPTION
**Pull request by hensldm**
Loadfragment2 OK and documented (basically the same as over in MM, with `bssSize` in the `OverlayRelocationSection` being an `s32` the only difference). As part of documenting brought over some headers to make things smoother.

Noticed a few typos in the yaml so changed
    * ../libultra/conteepwrite -> ../libultra/io/conteepwrite
    * ..libultra/libc/xprintf -> ../libultra/libc/xprintf
    * ./libultra/io/leodiskinit -> ../libultra/io/leodiskinit

Added a diff_settings.py

### CODE REVIEW

File: src/main/loadfragment2.c [hunk 1]
```diff
@@ -1,7 +1,133 @@
...
+                    *relocDataP = (*relocDataP & 0xFFFF0000) | (relocatedAddress & 0xFFFF);
+                } else if (gLoad2LogSeverity >= 3) {
+                }
+                break;
+        }
+    }
+}
+
+size_t func_8008169C(uintptr_t vRomStart, uintptr_t vRomEnd, uintptr_t vRamStart, uintptr_t vRamEnd,
+                     void* allocatedVRamAddr) {
```
Comment (AngheloAlf) [inline src/main/loadfragment2.c:91]: This function looks like `Load2_LoadOverlay`.
If you don't want to name it yet you could add a comment noting it is the same as the MM one

File: src/main/loadfragment2.c [hunk 2]
```diff
@@ -1,7 +1,133 @@
...
+
+    osWritebackDCache(allocatedVRamAddr, size);
+    osInvalICache(allocatedVRamAddr, size);
+
+    if (gLoad2LogSeverity >= 3) {}
+
+    return size;
+}
+
+void* func_80081744(uintptr_t vRomStart, uintptr_t vRomEnd, uintptr_t vRamStart, uintptr_t vRamEnd) {
```
Comment (AngheloAlf) [inline src/main/loadfragment2.c:125]: Same here for `Load2_AllocateAndLoad`

File: include/loadfragment2.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#ifndef LOADFRAGMENT2_H
```
Comment (AngheloAlf) [inline include/loadfragment2.h:1]: Oh, I like this header. We could consider adding it for MM and OoT too :eyes:

**Review by AngheloAlf — APPROVED** [review]

File: src/main/loadfragment2.c (hunk 1 above)
Comment (hensldm) [reply src/main/loadfragment2.c]: Ah, forgot 😅. Thanks for the reminder.

File: include/loadfragment2.h (hunk 3 above)
Comment (hensldm) [reply include/loadfragment2.h:1]: I've added it to my load PR in MM as `z64load.h`. As for OOT I'm not sure, I think they are waiting to split headers?

**Comment by ethteck** [comment]
Thanks! Lgtm :)

**Pull request merged**